### PR TITLE
parseUnverified returns cloudError

### PR DESCRIPTION
**Pull request by carvalhe**
### Which issue this PR addresses:

Fixes https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/15008430

### What this PR does / why we need it:

This PR is intended to have errors returned from parseUnverified in aad specifically to return the same cloudError as serviceProviderCredentials. As of now, we see logs where it keeps retrying to get token and the error "Invalid client secret provided." is displayed for 10 minuets only for it to go through and have invalid segments on a token. This change will give a better idea of the issue and split it from an internalServerError to a userError.

### CODE REVIEW

**Review by SudoBrendan — DISMISSED** [review]
lgtm

File: pkg/util/aad/aad.go [hunk 1]
```diff
@@ -57,6 +58,7 @@ func GetToken(ctx context.Context, log *logrus.Entry, clientID, clientSecret, te
 		claims := jwt.MapClaims{}
 		_, _, err = p.ParseUnverified(authorizer.OAuthToken(), claims)
 		if err != nil {
+			err = errServicePrincipal
```
Comment (weinong) [inline pkg/util/aad/aad.go:61]: i am not sure if this level of simplification is needed. I personally prefer to return a new cloud error here with more detail from `err`

**Review by ross-bryan — APPROVED** [review]
lgtm

**Review by s-amann — APPROVED** [review]

**Comment by weinong** [comment]
LGTM

**Pull request merged**